include/zephyr/drivers/sensor.h

```diff
@@ -540,12 +582,24 @@ struct sensor_read_config {
 	static enum sensor_channel __channel_array_##name[] = {__VA_ARGS__};                       \
 	static struct sensor_read_config __sensor_read_config_##name = {                           \
 		.sensor = DEVICE_DT_GET(dt_node),                                                  \
+		.is_streaming = false,                                                             \
 		.channels = __channel_array_##name,                                                \
 		.count = ARRAY_SIZE(__channel_array_##name),                                       \
 		.max = ARRAY_SIZE(__channel_array_##name),                                         \
 	};                                                                                         \
 	RTIO_IODEV_DEFINE(name, &__sensor_iodev_api, &__sensor_read_config_##name)
 
+#define SENSOR_DT_STREAM_IODEV(name, dt_node, ...)                                                 \
```

Comment:
Assumption: I think this isn't a super common thing (i.e. many sensors won't have this filter).

Overall, that feature sounds like we should be adding an attribute. This way the application can configure the FIFO to filter gyroscope data, and the steam request can focus on the when aspect. can you link a datasheet so I can see this filter?

Reply:
icm42688, icm20469, lsm6dso, bmi270 all support this sort of thing where you can select what goes in the fifo

Reply:
OK, found it in `FIFO_CONFIG1` for the icm42688, thanks. I still think this belongs more as an attribute than as a part of the read request no?

Reply:
I guess that's an interesting question here, what constitutes the stream options (sensor read config) vs sensor attributes. At the moment the poll interface lets you select channels and in this PR the stream interface lets you select which triggers but not which channels.

That seems a little strange I guess. I'm not opposed to making them attributes, but I guess if the goal is uniform interfaces for simplicity, it would make sense to allow the stream to also select which channels to read.

Reply:
I kind of agree with Tom.

Reply:
The problem I'm having with specifying the channels in this request is ambiguity. Now, please note I'm very possibly wrong here, but the point I'm making is that it's possible to make the following assumption because that's what I would think.

If I were implementing a driver and I got a steam request for the watermark trigger and just accel (no gyro). I would assume that means I can turn off the gyro. Now, the discussion we're having says no, just remove it from the FIFO. But that's not really specified in the API and it's open for interpretation.

Alternatively, if we add a new attribute `SENSOR_ATTR_FIFO_ACCEL_ENABLE` and the same for gyro, then there's no confusion. The steam option just says enable these triggers and give me any associated data. You'll always get the triggers with the timestamp, and it gives you 3 options of what to do with the rest of the data:
* Discard / flush
* Ignore / do nothing
* Read

There's a very clear line between what the API is trying to do. Anything more, you need to call sensor_set_attr